### PR TITLE
optimize:optimize sqlType actuator based on static factory pattern

### PR DESCRIPTION
**Pull request by pengzhengfa**

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
  * The type Execute template.
  *
  * @author sharajava
+ * @author pengzhengfa
```
Comment (a364176773) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java:30]: not a class you created, no need to add your author name

**Comment by a364176773** [comment]
there is no difference with the previous pr content, you can close one of them

**Comment by pengzhengfa** [comment]
@a364176773 Hello, there is something wrong with my warehouse, and then I submitted a new one

**Comment by pengzhengfa** [comment]
@l81893521 Can you help me with the audit

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java (hunk 1 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java:30]: I agree with @a364176773, change the author name only depends on create a new class or refactor the code structure.

**Comment by a364176773** [comment]
please resolve code conflicts

**Comment by pengzhengfa** [comment]
@a364176773 Conflict resolved

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java [hunk 2]
```diff
@@ -70,38 +68,16 @@
...
+            sqlRecognizer = SQLVisitorFactory.get(
                     statementProxy.getTargetSQL(),
                     statementProxy.getConnectionProxy().getDbType());
         }
         Executor<T> executor;
-        if (CollectionUtils.isEmpty(sqlRecognizers)) {
+        if (sqlRecognizer == null) {
             executor = new PlainExecutor<>(statementProxy, statementCallback);
         } else {
-            if (sqlRecognizers.size() == 1) {
```
Comment (l81893521) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java:83]: CI failed. Please check the code.

**Comment by pengzhengfa** [comment]
@l81893521 Hello, it's settled

**Comment by slievrly** [comment]
@pengzhengfa the branch is out of date，pls merge the latest branch.

**Comment by pengzhengfa** [comment]
@slievrly Hello, the merger has been completed, please check

**Pull request closed without merging**